UIDS-GIDS.md

```diff
@@ -0,0 +1,246 @@
+# Users, Groups, UIDs and GIDs on `systemd` systems
+
+Here's a summary of the requirements `systemd` (and Linux) make on UID/GID
+assignments and their ranges.
+
+Note that while in theory UIDs and GIDs are orthogonal concepts they really
+aren't IRL. With that in mind, when we discuss UIDs below it should be assumed
+that whatever we say about UIDs applies to GIDs in mostly the same way, and all
+the special assignments and ranges for UIDs always have mostly the same
+validity for GIDs too. It is recommended that every allocated system user
+should have a matching group, and they should share the same numeric value for
+their UIDs and GIDs. That said, while users/UIDs generally should have matching
+group/GID assigments, the reverse is not true the same way: there are plenty
+groups/GIDs that have no user counterpart.
```

Comment:
Can we please just delete the sentence about numeric values? IMO even "should" is completely unnecessary. If this catches on we'll have people "fixing" their installations by aligning the user/group numberings. At least Fedora, Debian, Ubuntu, and Arch have some system groups allocated by default and a mismatch in numbering. You even say "there are plenty of groups that have no user counterpart", which of course means that soon you'll have a mismatch in numbering.

Also "It is recommended that every allocated system user should have a matching group" — that's dubious. I mean that's the usual case, but there are many valid exceptions.